test/common/router/config_impl_test.cc

```diff
@@ -2399,11 +2400,47 @@ TEST_F(RouterMatcherHashPolicyTest, HashHeaders) {
   }
   {
     Http::TestRequestHeaderMapImpl headers = genHeaders("www.lyft.com", "/foo", "GET");
-    headers.addCopy("foo_header", "bar");
+    headers.addCopy("foo_header", "bar,");
     Router::RouteConstSharedPtr route = config().route(headers, 0);
     EXPECT_TRUE(route->routeEntry()->hashPolicy()->generateHash(nullptr, headers, add_cookie_nop_,
                                                                 nullptr));
   }
+  {
```

Comment:
for readability, can you factor out a helper-method for creating the header and hashing it?  Then the 4 stanzas you have here can look something like:
```
  absl::Hash<const std::vector<absl::string_view>> hasher;
  EXPECT_EQ(generateHash({"foo", "bar"}), generateHash({"bar", "foo"});
  EXPECT_EQ(generateHash({"foo", "bar"}), hasher({"foo", "bar"});
  EXPECT_EQ...
  EXPECT_NE
  ...
```

Reply:
got it, thanks for all the reviews!